.github/workflows/main.yml

```diff
@@ -227,13 +227,20 @@ jobs:
           flags: integration
 
   integration-report:
+    if: ${{ always() }}
     name: Integration Test Suite
     runs-on: ubuntu-latest
     needs: integration
     steps:
-      - name: "[Notification] Integration test suite passes"
+      - name: "Integration Tests Failed"
+        if: ${{ contains(needs.integration.result, 'failure') || contains(needs.integration.result, 'cancelled') }}
         run: |
-          echo "::notice title="Integration test suite passes""
+          echo "::notice title='Integration test suite failed'"
+          exit 1
```

Comment:
Does this mean next step won't be triggered if this step is executed? Might be good to have a comment since it is not really straightforward logic